lib/fsm/utils.go

```diff
@@ -41,9 +41,54 @@ func CanRollback(plan *storage.OperationPlan, phaseID string) error {
 		return trace.BadParameter(
 			"phase %q has already been rolled back", phase.ID)
 	}
+
+	leafPhases := plan.GetLeafPhases()
+
+	// TODO: Rollback of non-leaf phases may currently not be implemented
+	// properly. Roll back starts top-down, and not in reverse order.
+	if !containsPhase(leafPhases, phaseID) {
+		return trace.BadParameter(
+			"attempting to rollback non-leaf phase %q", phase.ID)
```

Comment:
Nit: I don't think a customer would understand why they receive an error like this. So I would consider going something more along the lines of:

`return trace.BadParameter("unable to rollback non-leaf phase, only leaf phases can be rolled back.").AddField("phase", phase.ID)`